src/transaction.ts

```diff
@@ -234,6 +234,10 @@ export class Transaction implements ISignable {
     serializeForSigning(signedBy: Address): Buffer {
         // TODO: for appropriate tx.version, interpret tx.options accordingly and sign using the content / data hash
         let plain = this.toPlainObject(signedBy);
+        // Make sure we never sign the transaction with another signature set up (useful when using the same method for verification)
+        if (plain.signature) {
```

Comment:
shouldn't this be part of `toPlainObject()` ?

Reply:
no, we should be able to convert `toPlainObject` while still keeping the signature. The sig removal should only be done while verifying the signature